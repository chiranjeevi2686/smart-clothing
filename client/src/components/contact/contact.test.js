import React from "react";
import { render, fireEvent } from '@testing-library/react';
import ContactContainer from './contact.component';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const defaultStore = 
    {
        displayName: '',
        name: '',
        email: '',
        message: ''
    };
const mockStore = configureStore();
describe('Collection contact component', () => {
    it('should render contact details', async () => {
        const { asFragment, getByTestId, getByText } = render(<Provider store={mockStore(defaultStore)}>
            <ContactContainer />
        </Provider>)
        expect(asFragment()).toMatchSnapshot();
        const displayNameInput = getByTestId('display-name');
        fireEvent.change(displayNameInput, {target: {value:'testDisplay'}})
        expect(displayNameInput.value).toEqual('testDisplay')

    })
})
