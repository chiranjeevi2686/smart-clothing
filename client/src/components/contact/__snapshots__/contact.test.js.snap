// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collection contact component should render contact details 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ eNxuqQ"
  >
    <h2
      class="sc-gsTCUz epPJTi"
    >
      Get in touch.
    </h2>
    <form>
      <div
        class="sc-dlfnbm tXewj"
      >
        <input
          class="sc-hKgILt puxKR"
          data-test-id="display-name"
          name="displayName"
          required=""
          type="text"
          value=""
        />
        <label
          class="sc-eCssSg iCNWHH"
        >
          Display Name
        </label>
      </div>
      <div
        class="sc-dlfnbm tXewj"
      >
        <input
          class="sc-hKgILt puxKR"
          name="email"
          required=""
          type="text"
          value=""
        />
        <label
          class="sc-eCssSg iCNWHH"
        >
          email
        </label>
      </div>
      <div
        class="sc-dlfnbm tXewj"
      >
        <input
          class="sc-hKgILt puxKR"
          name="message"
          required=""
          type="text"
          value=""
        />
        <label
          class="sc-eCssSg iCNWHH"
        >
          message
        </label>
      </div>
      <button
        class="sc-jSgupP wXXsH"
        type="submit"
      >
        SEND
      </button>
    </form>
  </div>
</DocumentFragment>
`;
